backend/internal/controller/server/msg_handler.go

```diff
@@ -52,14 +52,18 @@ func (s server) createResponse(reader io.Reader) ([]byte, error) {
 		return nil, err
 	}
 	switch message.Category {
-	case adapter.Initial:
+	case adapter.MsgInitial:
 		return json.Marshal(adapter.NewDispatcherResponse(message.TableID))
-	case adapter.Goal:
+	case adapter.MsgGoal:
 		s.game.AddGoal(message.Team)
 		return nil, nil
 	default:
-		log.Println("TableMessagesHandler: Bad message")
+		return nil, BadMessageError{}
 	}
+}
+
+type BadMessageError struct{}
 
-	return nil, nil
+func (err BadMessageError) Error() string {
+	return "msg_handlers/createResponse(): Unrecognized message type"
```

Comment:
Simulated stack trace is a bad idea

You might want to use [`fmt.Errorf`](https://pkg.go.dev/fmt#Errorf) for now, instead of custom error struct, since it's the simplest clean solution. Struct is nice, but you don't really need to complicate it like that for now ;)

Reply:
Changed